src/function/matrix/eigs.js

```diff
@@ -27,351 +48,168 @@ export const createEigs = /* #__PURE__ */ factory(name, dependencies, ({ typed,
-      Aki[k] = subtract(multiplyScalar(c, Hij[i][k]), multiplyScalar(s, Hij[j][k]))
-      Akj[k] = addScalar(multiplyScalar(s, Hij[i][k]), multiplyScalar(c, Hij[j][k]))
-    }
-    // Modify Hij
-    Hij[i][i] = Aii
-    Hij[j][j] = Ajj
-    Hij[i][j] = 0
-    Hij[j][i] = 0
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      if (k !== i && k !== j) {
-        Hij[i][k] = Aki[k]
-        Hij[k][i] = Aki[k]
-        Hij[j][k] = Akj[k]
-        Hij[k][j] = Akj[k]
+    for (let i = 0; i < N; i++) {
+      for (let j = 0; j < N; j++) {
+        const el = arr[i][j]
+
+        if (isNumber(el) || isFraction(el)) {
+          hasNumber = true
+        } else if (isBigNumber(el)) {
+          hasBig = true
+        } else if (isComplex(el)) {
+          hasComplex = true
+        } else {
+          throw TypeError('Unsupported type in Matrix: ' + typeOf(el))
+        }
       }
     }
-    return Hij
-  }
 
-  // update matrix
-  function x1 (Hij, theta, i, j) {
-    const N = Hij.length
-    const c = Math.cos(theta)
-    const s = Math.sin(theta)
-    const c2 = c * c
-    const s2 = s * s
-    const Aki = createArray(N, 0)
-    const Akj = createArray(N, 0)
-    //  Aii
-    const Aii = c2 * Hij[i][i] - 2 * c * s * Hij[i][j] + s2 * Hij[j][j]
-    const Ajj = s2 * Hij[i][i] + 2 * c * s * Hij[i][j] + c2 * Hij[j][j]
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      Aki[k] = c * Hij[i][k] - s * Hij[j][k]
-      Akj[k] = s * Hij[i][k] + c * Hij[j][k]
-    }
-    // Modify Hij
-    Hij[i][i] = Aii
-    Hij[j][j] = Ajj
-    Hij[i][j] = 0
-    Hij[j][i] = 0
-    // 0  to i
-    for (let k = 0; k < N; k++) {
-      if (k !== i && k !== j) {
-        Hij[i][k] = Aki[k]
-        Hij[k][i] = Aki[k]
-        Hij[j][k] = Akj[k]
-        Hij[k][j] = Akj[k]
-      }
+    if (hasBig && hasComplex) {
+      console.warn('Complex BigNumbers not supported, this operation will lose precission.')
     }
-    return Hij
-  }
 
-  // get max off-diagonal value from Upper Diagonal
-  function getAij (Mij) {
-    const N = Mij.length
-    let maxMij = 0
-    let maxIJ = [0, 1]
-    for (let i = 0; i < N; i++) {
-      for (let j = i + 1; j < N; j++) {
-        if (Math.abs(maxMij) < Math.abs(Mij[i][j])) {
-          maxMij = Math.abs(Mij[i][j])
-          maxIJ = [i, j]
+    if (hasComplex) {
+      for (let i = 0; i < N; i++) {
+        for (let j = 0; j < N; j++) {
+          arr[i][j] = complex(arr[i][j])
         }
       }
+
+      return 'Complex'
     }
-    return [maxIJ, maxMij]
-  }
 
-  // get max off-diagonal value from Upper Diagonal
-  function getAijBig (Mij) {
-    const N = Mij.length
-    let maxMij = 0
-    let maxIJ = [0, 1]
-    for (let i = 0; i < N; i++) {
-      for (let j = i + 1; j < N; j++) {
-        if (abs(maxMij) < abs(Mij[i][j])) {
-          maxMij = abs(Mij[i][j])
-          maxIJ = [i, j]
+    if (hasBig) {
+      for (let i = 0; i < N; i++) {
+        for (let j = 0; j < N; j++) {
+          arr[i][j] = bignumber(arr[i][j])
```

Comment:
The function `bignumber` also accepts array or matrix input, so you can replace the two for loops simply with
`arr = bignumber(arr)`?